File: 전예진/Graph/prob1.py
from collections import deque

def solution(n, edge):
    graph = {i: [] for i in range(1, n+1)}

    for e in edge:
        a, b = e
        graph[a].append(b)
        graph[b].append(a)

    def bfs(start):
        visited = [False] * (n + 1)
        queue = deque([(start, 0)])
        max_distance = 0
        count = 0

        while queue:
            node, distance = queue.popleft()

            if not visited[node]:
                visited[node] = True

                if distance > max_distance:
                    max_distance = distance
                    count = 1
                elif distance == max_distance:
                    count += 1

                for neighbor in graph[node]:
                    if not visited[neighbor]:
                        queue.append((neighbor, distance + 1))

        return count

    return bfs(1)

File: 주다애/0701/prob1.java
class Solution {
    public int solution(int n, int[] money) {
        int len = money.length;
        int[][] dp = new int[len + 1][n + 1];

        for (int i = 1; i <= len; i++) {
            for (int j = 0; j <= n; j++) {
                if (j == 0) {
                    dp[i][j] = 1;
                } else if (j - money[i - 1] >= 0) {
                    dp[i][j] = (dp[i - 1][j] + dp[i][j - money[i - 1]]) % 1000000007;
                } else
                    dp[i][j] = dp[i - 1][j];
            }
        }

        return dp[len][n];
    }
}
